sdk/python/feast/permissions/client/http_auth_requests_wrapper.py

```diff
@@ -0,0 +1,44 @@
+import requests
+
+from feast.permissions.auth_model import AuthConfig
+from feast.permissions.client.k8_authentication_client_manager import (
+    K8AuthClientManager,
+)
+from feast.permissions.client.oidc_authentication_client_manager import (
+    OidcAuthClientManager,
+)
+
+
+class AuthenticatedRequestsSession(requests.Session):
+    def __init__(self, auth_token: str):
+        super().__init__()
+        self.auth_token = auth_token
+        self.headers.update({"Authorization": f"Bearer {self.auth_token}"})
+
+
+class AuthClientManagerFactory:
+    @staticmethod
+    def get_auth_client_manager(auth_config: AuthConfig):
+        if auth_config.type == "oidc":
```

Comment:
should we also validate that the type of `auth_config` is consistent with the actual `auth_config.type`?

Reply:
Added the assert to type.